### PR TITLE
Add `core.batch()` API

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

**Comment by kasperisager** [comment]
This should ideally write blocks to disk rather than keep them in memory and then delay updating the tree until the blocks are committed. Might be for a separate PR though.

File: lib/batch.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    try {
+      if (this._appends.length) await this.session.append(this._appends)
+    } finally {
+      this._clearBatch()
+      this._clearAppends()
+    }
+  }
+
+  async destroy () {
```
Comment (mafintosh) [inline lib/batch.js:94]: can we name this close for parity with a session?

File: lib/batch.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+module.exports = class Batch {
+  constructor (session) {
+    this.session = session
+    this.flushed = false
+
+    this._appends = []
+    this._byteLength = 0
+  }
+
```
Comment (mafintosh) [inline lib/batch.js:11]: can we add ready () that just forwards to session.ready()

**Pull request merged**